packages/nextjs-mf/src/plugins/AddModulesToRuntime.js

```diff
@@ -29,25 +61,154 @@ class AddModulesToRuntimeChunkPlugin {
     const { runtime, container, remotes, shared, eager, applicationName } =
       this.options;
 
-    new DelegateModulesPlugin({
-      runtime,
-      container,
-      remotes,
-    }).apply(compiler);
+    isServer &&
+      new DelegateModulesPlugin({
+        runtime,
+        container,
+        remotes,
+      }).apply(compiler);
+
+    if (!isServer) {
+      // next-client-pages-loade
+    }
+    if (typeof compiler.options.entry === 'function' && !isServer) {
+      const backupEntries = compiler.options.entry;
+      compiler.options.entry = () =>
+        backupEntries().then((entries) => {
+          //loop over object with for
+          if (entries) {
+            for (const [key, value] of Object.entries(entries)) {
+              if (key === 'main') {
+                value.import[0] =
+                  require.resolve('./async-pages-loader') +
+                  '!' +
+                  value.import[0];
+              }
+              // if (key === 'pages/_app') {
+              //   value.import[1] =
+              //     require.resolve('./async-pages-loader') +
+              //     '!' +
+              //     value.import[1];
+              // }
+              if (value.import[0].startsWith('next-client-pages-loader')) {
+                value.import[0] =
+                  require.resolve('./async-pages-loader') +
+                  '!' +
+                  value.import[0];
+              }
+            }
+          }
+          return entries;
+        });
+    }
 
     // Tap into compilation hooks
     compiler.hooks.compilation.tap(
       'AddModulesToRuntimeChunkPlugin',
       (compilation) => {
-        return;
         // Tap into optimizeChunks hook
         compilation.hooks.optimizeChunks.tap(
           'AddModulesToRuntimeChunkPlugin',
           (chunks) => {
+            console.log('running optimize chunks action');
             // Get the runtime chunk and return if it's not found or has no runtime
-            const runtimeChunk = this.getChunkByName(chunks, runtime);
-            if (!runtimeChunk || !runtimeChunk.hasRuntime()) return;
+            const mainChunk = this.getChunkByName(chunks, 'main');
+            const runtimeChunk = this.getChunkByName(chunks, 'webpack');
```

Comment:
Initial experiments 🧪 
Move eager out of remote

Reply:
this didn't work on its own, i had to add a second module federation plugin instance to overcome hmr share scope resetting when lazy compile happens.